client/state/editor/reducer.js

```diff
@@ -0,0 +1,45 @@
+/**
+ * External dependencies
+ */
+import { combineReducers } from 'redux';
+
+/**
+ * Internal dependencies
+ */
+import {
+	EDITOR_POST_EDIT,
+	SERIALIZE,
+	DESERIALIZE
+} from 'state/action-types';
+
+/**
+ * Returns the updated editor posts state after an action has been dispatched.
+ * The state maps site ID, post ID pairing to an object containing revisions
+ * for the post.
+ *
+ * @param  {Object} state  Current state
+ * @param  {Object} action Action payload
+ * @return {Object}        Updated state
+ */
+export function posts( state = {}, action ) {
+	switch ( action.type ) {
+		case EDITOR_POST_EDIT:
+			const { post, siteId, postId = '' } = action;
+			state = Object.assign( {}, state, {
+				[ siteId ]: Object.assign( {}, state[ siteId ] )
+			} );
+
+			state[ siteId ][ postId ] = Object.assign( {}, state[ siteId ][ postId ], post );
```

Comment:
>Do we need cloneDeep here, or is the post object one level deep?
 
All but one property would be covered by a shallow merge: `discussion`. It's possible to save `discussion. comments_open` and `discussion.pings_open` separately, which would necessitate a deep merge. I'll explore options here and ensure that a test case is added.

Related docs: https://developer.wordpress.com/docs/api/1.1/post/sites/%24site/posts/%24post_ID/